### ISSUE TITLE
Issue deploying Azure RM MySQL Flexible Database

### ISSUE THREAD

**Issue by mattjames1978** [hashicorp/terraform-provider-azurerm#15322]
I am running some terraform code to create a new Flexible MySQL server and database which is fairly new to version 2.89 and above.
However I encounter the following error as if the server is not ready for the database to be added. I also have observed that it takes a round 16 minutes to provision a new MY SQL Flexible server. My code was fine when building a normal My SQL instance with DB.

here is a snippet of the code:

############ My SQL Flexible Server Configuration #################

resource "azurerm_mysql_flexible_server" "wordpress" {
  name                        = "${var.name}-mysqlserver"
  location                    = azurerm_resource_group.wordpress.location
  resource_group_name         = azurerm_resource_group.wordpress.name
  tags                        = var.tags
  sku_name                    = "GP_Standard_D2ds_v4"
  version                     = "8.0.21"
  #version                     = "5.7"
  administrator_login         = var.mysql_admin_user
  administrator_password      = random_password.mysql.result
  backup_retention_days       = 7
   high_availability {
     mode                      = "ZoneRedundant"
     standby_availability_zone = "2"
   }
   zone = "1"
   storage {
     auto_grow_enabled         = true
   }
}

resource "azurerm_mysql_flexible_database" "wordpress" {
   name                = "wordpressdb"
#   #name                = "${var.name}db"
   resource_group_name = azurerm_resource_group.wordpress.name
   server_name         = azurerm_mysql_flexible_server.wordpress.name
   charset             = "utf8"
   collation           = "utf8_unicode_ci"
 }

resource "azurerm_mysql_flexible_server_configuration" "wordpress" {
  name                = "interactive_timeout"
  resource_group_name = azurerm_resource_group.wordpress.name
  server_name         = azurerm_mysql_flexible_server.wordpress.name
  value               = "600"
}

resource "azurerm_mysql_flexible_server_firewall_rule" "azure" {
  name                = "public-internet"
  resource_group_name = azurerm_resource_group.wordpress.name
  server_name         = azurerm_mysql_flexible_server.wordpress.name
  start_ip_address    = "0.0.0.0"
  end_ip_address      = "255.255.255.255"
}

The error states Error: waiting on creation of Flexible Database: (Database Name "wordpressdb" / Flexible Server Name "wordpress-xyz-mysqlserver" /resource Group "wordpress-xyz-rg"): Code="ServerNotFound" Message="Subscription ****** does not have the server 'wordpress-xyz-mysqlserver'."

If I run apply a second time it runs without issue.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_XXXXX`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by neil-yechenwei** [comment]
@mattjames1978 , thanks for raising this issue. After tested with below tf config, seems I cannot repro this issue. Could you try below tf config with latest azurerm provider to see if the issue still exists? Thanks.

tf config:
```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-mysqlfsdb-test01"
  location = "West Europe"
}
resource "azurerm_mysql_flexible_server" "test" {
  name                   = "acctest-mysqlfs-test02"
  resource_group_name    = azurerm_resource_group.test.name
  location               = azurerm_resource_group.test.location
  administrator_login    = "adminTerraform"
  administrator_password = "HBZwsx123"
  sku_name               = "GP_Standard_D2ds_v4"
  version                = "8.0.21"
  backup_retention_days  = 7
  zone                   = "1"

  high_availability {
    mode                      = "ZoneRedundant"
    standby_availability_zone = "2"
  }

  storage {
    auto_grow_enabled = true
  }

  tags = {
    env = "test"
  }
}

resource "azurerm_mysql_flexible_database" "test" {
  name                = "acctestmysqlfsdbtest02"
  resource_group_name = azurerm_resource_group.test.name
  server_name         = azurerm_mysql_flexible_server.test.name
  charset             = "utf8"
  collation           = "utf8_unicode_ci"
}
```

**Comment by mattjames1978** [comment]
Hi Neil, thanks for looking into this for me.

That has really helped and it seems what is causing my issue is when I use a variable for the MYSQL random password string for the password. So If I hard code the password like you have done it runs without issue. Using the variable for the username seems to be ok from the testing I have done.

***@***.***

***@***.***

***@***.***

I have had a play around with the password length and reduced to 8 characters which seems to be what the issue was, the one I was setting was 64!!

Did you notice the creation time is around 16 minutes for this as well? Quite a while.

***@***.***

Many thanks
____________________________________________
Matt James

Azure Cloud Architect | Capgemini UK

Capgemini UK | Woking
Forge End, Woking, GU21 6DB
Mob.: + 44 7467892486
www.capgemini.com<http://www.capgemini.com/>

Email ***@***.******@***.***>
***@***.***
[A picture containing text  Description automatically generated]<https://www.capgemini.com/resources/get-the-future-you-want>
____________________________________________
Connect with Capgemini:
***@***.***<https://twitter.com/Capgemini> ***@***.*** <https://www.facebook.com/Capgemini/>  ***@***.*** <https://www.linkedin.com/company/capgemini/mycompany/verification/>  ***@***.*** <https://www.youtube.com/profile?user=capgeminimedia>  ***@***.*** <https://www.slideshare.net/capgemini>  ***@***.*** <https://soundcloud.com/capgemini>  ***@***.*** <https://www.instagram.com/capgemini/>  ***@***.*** <https://www.glassdoor.fr/Overview/Working-at-Capgemini-EI_IE3803.11,20.htm?countryRedirect=true>

Please consider the environment and do not print this email unless absolutely necessary.
Capgemini encourages environmental awareness.

**Comment by neil-yechenwei** [comment]
@mattjames1978 , as you enabled `high_availability`, so creating server takes a while. It's expected. 

After tested with `random_password` resource and set length to 8 and 64, I still cannot repro this issue. Suggest simply your tf config that triggers this issue and compare with below tf config to check what's the difference between them for troubleshooting? Thanks.

Below tf config that can create these resources successfully:
main.tf:
```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-mysqlfsdb-test02"
  location = "West Europe"
}

resource "random_password" "mysql" {
  length           = 64
  special          = true
  override_special = "_%@"
  lower            = true
  number           = true
  upper            = true
  min_lower        = 1
  min_numeric      = 1
  min_special      = 1
  min_upper        = 1
}

resource "azurerm_mysql_flexible_server" "test" {
  name                   = "acctest-mysqlfs-test02"
  resource_group_name    = azurerm_resource_group.test.name
  location               = azurerm_resource_group.test.location
  administrator_login    = var.username
  administrator_password = random_password.mysql.result
  sku_name               = "GP_Standard_D2ds_v4"
  version                = "8.0.21"
  backup_retention_days  = 7
  zone                   = "1"

  high_availability {
    mode                      = "ZoneRedundant"
    standby_availability_zone = "2"
  }

  storage {
    auto_grow_enabled = true
  }

  tags = {
    env = "test"
  }
}

resource "azurerm_mysql_flexible_database" "test" {
  name                = "acctestmysqlfsdbtest02"
  resource_group_name = azurerm_resource_group.test.name
  server_name         = azurerm_mysql_flexible_server.test.name
  charset             = "utf8"
  collation           = "utf8_unicode_ci"
}
```

variables.tf:
```
variable "username" {
  description = "user name"
  default     = "adminTerraform"
}
```

**Comment by neil-yechenwei** [comment]
@mattjames1978 , unfortunately, I still cannot repro this issue. Suggest simply your tf config that triggers this issue and compare with below tf config to check what's the difference between them to locate the root case? Thanks.

```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-mysqlfsdb-test08"
  location = "West Europe"
}

resource "random_password" "mysql" {
  length           = 64
  special          = true
  override_special = "_%@"
  lower            = true
  number           = true
  upper            = true
  min_lower        = 1
  min_numeric      = 1
  min_special      = 1
  min_upper        = 1
}

resource "azurerm_mysql_flexible_server" "test" {
  name                   = "acctest-mysqlfs-test08"
  resource_group_name    = azurerm_resource_group.test.name
  location               = azurerm_resource_group.test.location
  administrator_login    = var.username
  administrator_password = random_password.mysql.result
  sku_name               = "GP_Standard_D2ds_v4"
  version                = "8.0.21"
  backup_retention_days  = 7
  zone                   = "1"

  high_availability {
    mode                      = "ZoneRedundant"
    standby_availability_zone = "2"
  }

  storage {
    auto_grow_enabled = true
  }

  tags = {
    env = "test"
  }
}

resource "azurerm_mysql_flexible_server_firewall_rule" "test" {
  name                = "acctest-public-internet"
  resource_group_name = azurerm_resource_group.test.name
  server_name         = azurerm_mysql_flexible_server.test.name
  start_ip_address    = "0.0.0.0"
  end_ip_address      = "255.255.255.255"
}

resource "azurerm_mysql_flexible_server_configuration" "test" {
  name                = "interactive_timeout"
  resource_group_name = azurerm_resource_group.test.name
  server_name         = azurerm_mysql_flexible_server.test.name
  value               = "30"
}

resource "azurerm_mysql_flexible_database" "test" {
  name                = "acctestmysqlfsdbtest08"
  resource_group_name = azurerm_resource_group.test.name
  server_name         = azurerm_mysql_flexible_server.test.name
  charset             = "utf8"
  collation           = "utf8_unicode_ci"
}
```

**Comment by mattjames1978** [comment]
Hi Neil, ok thanks will do. Trying to deploy it with a word press docker container, web app but deployment seems to be behaving now.

thanks
____________________________________________
Matt James

Azure Cloud Architect | Capgemini UK

Capgemini UK | Woking
Forge End, Woking, GU21 6DB
Mob.: + 44 7467892486
www.capgemini.com<http://www.capgemini.com/>

Email ***@***.******@***.***>
***@***.***
[A picture containing text  Description automatically generated]<https://www.capgemini.com/resources/get-the-future-you-want>
____________________________________________
Connect with Capgemini:
***@***.***<https://twitter.com/Capgemini> ***@***.*** <https://www.facebook.com/Capgemini/>  ***@***.*** <https://www.linkedin.com/company/capgemini/mycompany/verification/>  ***@***.*** <https://www.youtube.com/profile?user=capgeminimedia>  ***@***.*** <https://www.slideshare.net/capgemini>  ***@***.*** <https://soundcloud.com/capgemini>  ***@***.*** <https://www.instagram.com/capgemini/>  ***@***.*** <https://www.glassdoor.fr/Overview/Working-at-Capgemini-EI_IE3803.11,20.htm?countryRedirect=true>

Please consider the environment and do not print this email unless absolutely necessary.
Capgemini encourages environmental awareness.

**Comment by rcskosir** [comment]
Thanks for opening this issue. This was a problem in the 2.x version of the provider which is no longer actively maintained.
If this is still an issue with the 3.x version of the provider please do let us know by opening a new issue, thanks!
